snuba/datasets/querylog.py

```diff
@@ -0,0 +1,89 @@
+from datetime import timedelta
+from snuba import settings
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnSet,
+    DateTime,
+    Enum,
+    Float,
+    LowCardinality,
+    Mapping,
+    Nested,
+    String,
+    UInt,
+)
+
+from snuba.datasets.dataset import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.schemas.tables import MergeTreeSchema
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.datasets.querylog_processor import QuerylogProcessor
+from snuba.query.extensions import QueryExtension
+from snuba.query.project_extension import ProjectExtension, ProjectExtensionProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+
+
+class QuerylogDataset(TimeSeriesDataset):
+    def __init__(self) -> None:
+        status_type = Enum([("success", 0), ("error", 1), ("rate-limited", 2)])
+
+        columns = ColumnSet(
+            [
+                ("request", String()),
+                ("referrer", LowCardinality(String())),
+                ("dataset", LowCardinality(String())),
+                ("projects", Array(UInt(64))),
+                ("timestamp", DateTime()),
+                ("duration_ms", UInt(32)),
+                ("status", status_type),
+                (
+                    "clickhouse_queries",
+                    Nested(
```

Comment:
I would add the timing columns in the clickhouse query as well even if you are not populating them yet, we know we will need them soon, so we avoid spending the time for the migration.

Reply:
Added duration_ms to the nested queries (I don't think anything else is particuarly useful)